Simplified VisualizationConverter methods for Execution conversion

Dominick1993 opened the pull request:
We wanted to convert objects of VisualizationObject and VisualizationClass that we obtained beforehand, but we were forced to write some boilerplate function that returned the VisualizationClass instead of providing the VisualizationClass directly. The modified methods simply allow to provide the VisualizationClass directly, while still keeping the backward compatibility with previous approach.
In addition, new check was implemented that checks whether the VisualizationClass URI matches with the URI that VisualizationObject contains. If you try to create convert incompatible objects, exception will be thrown.

Fixes #686

gdgate commented:
Build succeeded (check pipeline).

xMort commented on src/main/java/com/gooddata/md/visualization/VisualizationConverter.java:
```diff
@@ -77,11 +94,30 @@ public static Afm convertToAfm(final VisualizationObject visualizationObject) {
+    public static ResultSpec convertToResultSpec(final VisualizationObject visualizationObject,
+            final VisualizationClass visualizationClass) throws Exception {
+        if (!visualizationObject.getVisualizationClassUri().equals(visualizationClass.getUri())) {
+            throw new Exception("VisualizationClass URI does not match the VisualizationObject URI, "
```
Wouldn't it be nicer to have some specific exception type instead?

Dominick1993 replied:
It would be for sure, wanted to discuss this!

gdgate commented:
Build succeeded (check pipeline).

Dominick1993 commented:
Now using `Validate.isTrue` that throws `IllegalArgumentException`, fixed failing tests and introduced one more regarding the `IllegalArgumentException`

gdgate commented:
Build succeeded (check pipeline).

gdgate commented:
Build succeeded (check pipeline).

peter-plochan commented on src/test/groovy/com/gooddata/md/visualization/VisualizationConverterTest.groovy:
```diff
@@ -160,4 +170,15 @@ class VisualizationConverterTest extends Specification {
+    def "should throw IllegalArgumentException"() {
+        when:
+        convertToResultSpec(
+                Stub(VisualizationObject) { getVisualizationClassUri() >> "visClassUri"},
```
```suggestion
                Stub(VisualizationObject) { getVisualizationClassUri() >> 'visClassUri'},
```

peter-plochan commented on src/test/groovy/com/gooddata/md/visualization/VisualizationConverterTest.groovy:
```diff
@@ -160,4 +170,15 @@ class VisualizationConverterTest extends Specification {
+        when:
+        convertToResultSpec(
+                Stub(VisualizationObject) { getVisualizationClassUri() >> "visClassUri"},
+                Stub(VisualizationClass) { getUri() >> "nonMatchingVisClassUri"}
```
```suggestion
                Stub(VisualizationClass) { getUri() >> 'nonMatchingVisClassUri'}
```

peter-plochan commented on src/test/groovy/com/gooddata/md/visualization/VisualizationConverterTest.groovy:
```diff
@@ -160,4 +170,15 @@ class VisualizationConverterTest extends Specification {
         properties << ['abcd', '{"sortItems":[{"someinvalidstring"}]}']
     }
+
+    def "should throw IllegalArgumentException"() {
```
```suggestion
    def "should fail when incorrect visualization class provided"() {
```

peter-plochan commented on src/main/java/com/gooddata/md/visualization/VisualizationConverter.java:
```diff
@@ -80,9 +94,24 @@ public static Afm convertToAfm(final VisualizationObject visualizationObject) {
+    public static ResultSpec convertToResultSpec(final VisualizationObject visualizationObject,
+            final VisualizationClass visualizationClass) {
+        Validate.isTrue(visualizationObject.getVisualizationClassUri().equals(visualizationClass.getUri()),
+                "VisualizationClass URI does not match the VisualizationObject URI, "
```
```suggestion
                "visualizationClass URI does not match the URI within visualizationObject, "
```

peter-plochan commented:
additionally, could you please add `notNull` checks to all (even not changed) public conversion methods (whenever it makes sense)? thanks :)

gdgate commented:
Build succeeded (check pipeline).

gdgate commented:
Build succeeded (check pipeline).

gdgate commented:
Build succeeded (check pipeline).

gdgate commented:
Build succeeded (check pipeline).
